src/models/parallelCoordinatesChart.js

```diff
@@ -0,0 +1,331 @@
+                // Setup containers and skeleton of chart
+
+                var wrap = container.selectAll('g.nv-wrap.nv-parallelCoordinatesChart').data([data]);
+                var gEnter = wrap.enter().append('g').attr('class', 'nvd3 nv-wrap nv-parallelCoordinatesChart').append('g');
+
+                var g = wrap.select('g');
+                
+                gEnter.append('g').attr('class', 'nv-parallelCoordinatesWrap');
+                gEnter.append('g').attr('class', 'nv-legendWrap');
+                gEnter.append('g').attr('class', 'nv-controlsWrap');
+
+                g.select("rect")
+                    .attr("width", availableWidth)
+                    .attr("height", (availableHeight > 0) ? availableHeight : 0);
+
+                // Legend
+                if (showLegend) {
+                    legend.width(availableWidth - controlWidth())
+                        .color(function (d) { return "rgb(188,190,192)"; });
+
+                    g.select('.nv-legendWrap')
+                        .datum(dimensionNames.sort(function (a, b) { return a.originalPosition - b.originalPosition; }))
+                        .call(legend);
+
+                    if (margin.top != legend.height()) {
+                        margin.top = legend.height();
+                        availableHeight = nv.utils.availableHeight(height, container, margin);
+                    }
+                    wrap.select('.nv-legendWrap')
+                       .attr('transform', 'translate(' + controlWidth() + ',' + (-margin.top) + ')');
+                }
+                wrap.attr('transform', 'translate(' + margin.left + ',' + margin.top + ')');
+
+                // Controls
+                if (showControls) {
+                    if (resetBrushButton === null) {
+                        resetBrushButton = d3.select(container.node().parentNode).append("button")
+                            .attr("class", "nv-resetBrushButton")
+                            .text("Reset Brush");
+
+                        resetBrushButton.on('click', function () {
+                            parallelCoordinates.filters([]);
+                            parallelCoordinates.active([]);
+                            parallelCoordinates.displayBrush(true);
+                            dispatch.resetBrush(data);
+
+                            parallelCoordinatesWrap.transition().call(parallelCoordinates);
+                            resetBrushButton.style("visibility", "hidden");
+                        });
+
+                        resetBrushButton.node().parentNode.insertBefore(container.node(), resetBrushButton.node().nextSibling);
+                    }
+
+                    if (resetSortingButton === null) {
+                        resetSortingButton = d3.select(container.node().parentNode).append("button")
+                            .attr("class", "nv-resetSortingButton")
+                            .text("Reset Sorting");
```

Comment:
Why are we including button controls in the actual code?    This should probably be put in the example HTML and then affects the chart either through setting changes or dispatch.